### PR TITLE
Feature: Allow `yml` extension

### LINKED ISSUE #161: Allow `.yml` as YAML file extension
### Issue Summary

When defining template context files (the YAML) files, these are only recognised when they use the full `.yaml` extension. 

When using the shorter `.yml` extension the template is not picking up the context file. 

This can be very confusing. 

### Steps to Reproduce

1. Setup a project with `django-patter-library` enable
2. Create a template `test.html` in one of the directories observed by `django-pattern-library` with the content 

 ```django
{{ test_var }}
````

3. Create a context file `test.yaml` next to the template with the content

```yaml
context:
  test_var: Test content
````

4. Open the template in the patter library and observe that you see "Test content"
5. Rename the context file `mv test.yaml test.yml`
6. Open the template in the patter library and observe that you see nothing. The context is not rendered anymore.

### PR DESCRIPTION
**Pull request by tbrlpld**
## Description

So far, the context YAML files had to have the `.yaml` extension. YAML files often also use the `.yml` extension. This PR adds a second attempt at finding the YAML file with the `.yml` extension after the `.yaml` extension has failed to produce a result. 

The second attempt is simply enabled by adding a second nested try-except-block. I am not sure this is the most elegant solution, but it works and seems easy to understand. 

Fixes #161 

## Checklist

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas -> No hard to understand areas. 
- [x] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] I have added an appropriate CHANGELOG entry

### CODE REVIEW

**Review by thibaudcolas — COMMENTED** [review]
@tbrlpld this seems spot on to me – we have to attempt to read both paths either way, so I don’t think there would be much of a better way. I have a feeling this is overkill but we could cache whichever file extension was loaded successfully last for a given pattern to avoid the extra IO when a site consistently uses `.yml`?

I’d additionally want to see:

- A unit test for this
- A mention in the documentation, replacing the "Always use `.yaml`" we have here: https://torchbox.github.io/django-pattern-library/reference/api/#yaml-structure

**Comment by tbrlpld** [comment]
@thibaudcolas I have pushed changes according to your review. 

Regarding the caching: I also think that would be overkill. Especially in larger projects I can imagine that both extensions could occur when no linter or CI checks preventing it from happening. In that case it would be particularly confusing why the files are not recognized while they would work in another project which is consistent. 

If we wanted to allow projects to be consistent, we could add a setting, but I am afraid that could also cause confusion because you would need to know how it project is configured. 

I am not sure, but would using `glob` to check for file existence save the additional IO?

**Comment by thibaudcolas** [comment]
@tbrlpld thank you, I’ll try to review this once I get the chance. Re `glob` – I doubt it’s worth it. We’d have to do 1 `scandir` with glob, then the 1 file read, so it’d be two IO calls for all cases rather than the "1 or 2" situation we’d be in here.

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,8 @@
 
 ## [Unreleased]
 
+- Allow `.yml` extension for YAML files. ([#161](https://github.com/torchbox/django-pattern-library/issues/161))
```
Comment (bcdickinson) [inline CHANGELOG.md:5]: This should be under a heading as per https://keepachangelog.com/en/1.0.0/ 

I think "Changed" is the most appropriate here.

**Review by bcdickinson — APPROVED** [review]
Thanks Tibor, this is great.

**Comment by tbrlpld** [comment]
Thanks for reviewing @bcdickinson

**Pull request merged**